main.tf

```
@@ -75,10 +75,10 @@ resource "azurerm_firewall_policy_rule_collection_group" "this" {
 
       rule {
         name                  = "allow-dc-to-dnsresolver-inbound"
-        protocols             = ["Any"]
+        protocols             = ["UDP"]
```

As far as I know, DNS over TCP is required for a DNS server.
```suggestion
        protocols             = ["UDP","TCP"]
```